haystack/preview/document_stores/memory/document_store.py

```diff
@@ -23,7 +23,7 @@
 SCALING_FACTOR = 8
 
 
-class MemoryDocumentStore:
+class MemoryDocumentStore(Store):
```

Comment:
Even though this is not an error it's not necessary to inherit from a `Protocol` if it doesn't have any default implementations already. There's some more info in [PEP 554](https://peps.python.org/pep-0544/#explicitly-declaring-implementation).

Since we're using `Store` to just define the expect interface of all `Store`s I would just omit this.